homeassistant/components/sabnzbd.py

```diff
@@ -0,0 +1,250 @@
+    CONF_HOST, CONF_API_KEY, CONF_NAME, CONF_PORT, CONF_SENSORS,
+    CONF_SSL, CONF_SCAN_INTERVAL)
+from homeassistant.helpers import discovery
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pysabnzbd==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'sabnzbd'
+DATA_SABNZBD = 'sabznbd'
+
+_CONFIGURING = {}
+
+ATTR_SPEED = 'speed'
+BASE_URL_FORMAT = '{}://{}:{}/'
+CONFIG_FILE = 'sabnzbd.conf'
+DEFAULT_HOST = 'localhost'
+DEFAULT_NAME = 'SABnzbd'
+DEFAULT_PORT = 8080
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+DEFAULT_SSL = False
+
+SERVICE_PAUSE = 'pause'
+SERVICE_RESUME = 'resume'
+SERVICE_SET_SPEED = 'set_speed'
+
+SIGNAL_SABNZBD_UPDATED = 'sabnzbd_updated'
+
+SENSOR_TYPES = {
+    'current_status': ['Status', None, 'status'],
+    'speed': ['Speed', 'MB/s', 'kbpersec'],
+    'queue_size': ['Queue', 'MB', 'mb'],
+    'queue_remaining': ['Left', 'MB', 'mbleft'],
+    'disk_size': ['Disk', 'GB', 'diskspacetotal1'],
+    'disk_free': ['Disk Free', 'GB', 'diskspace1'],
+    'queue_count': ['Queue Count', None, 'noofslots_total'],
+    'day_size': ['Daily Total', 'GB', 'day_size'],
+    'week_size': ['Weekly Total', 'GB', 'week_size'],
+    'month_size': ['Monthly Total', 'GB', 'month_size'],
+    'total_size': ['Total', 'GB', 'total_size'],
+}
+
+SPEED_LIMIT_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_SPEED): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+            cv.time_period_seconds,
+        vol.Optional(CONF_SENSORS):
+            vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_check_sabnzbd(sab_api):
+    """Check if we can reach SABnzbd."""
+    from pysabnzbd import SabnzbdApiException
+
+    try:
+        await sab_api.check_available()
+        return True
+    except SabnzbdApiException:
+        _LOGGER.error("Connection to SABnzbd API failed")
+        return False
+
+
+async def async_configure_sabnzbd(hass, config, use_ssl, name=DEFAULT_NAME,
+                                  api_key=None):
+    """Try to configure Sabnzbd and request api key if configuration fails."""
+    from pysabnzbd import SabnzbdApi
+
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+    uri_scheme = 'https' if use_ssl else 'http'
+    base_url = BASE_URL_FORMAT.format(uri_scheme, host, port)
+    if api_key is None:
+        conf = load_json(hass.config.path(CONFIG_FILE))
```

Comment:
You cannot do I/O inside an async context 😮 

Yield to the executor: `conf = await hass.async_add_job(load_json, hass.config.path(CONFIG_FILE)`

Same goes for writing.